spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/StreamsBuilderFactoryManager.java

```diff
@@ -81,19 +89,24 @@ public synchronized void start() {
 			try {
 				Set<StreamsBuilderFactoryBean> streamsBuilderFactoryBeans = this.kafkaStreamsBindingInformationCatalogue
 						.getStreamsBuilderFactoryBeans();
-				int n = 0;
 				for (StreamsBuilderFactoryBean streamsBuilderFactoryBean : streamsBuilderFactoryBeans) {
 					if (this.listener != null) {
 						streamsBuilderFactoryBean.addListener(this.listener);
 					}
-					// By default, we shutdown the client if there is an uncaught exception in the application.
+					// By default, we shut down the client if there is an uncaught exception in the application.
 					// Users can override this by customizing SBFB. See this issue for more details:
 					// https://github.com/spring-cloud/spring-cloud-stream-binder-kafka/issues/1110
 					streamsBuilderFactoryBean.setStreamsUncaughtExceptionHandler(exception ->
 							StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse.SHUTDOWN_CLIENT);
 					// Starting the stream.
-					streamsBuilderFactoryBean.start();
-					this.kafkaStreamsRegistry.registerKafkaStreams(streamsBuilderFactoryBean);
+					final Map<StreamsBuilderFactoryBean, List<ConsumerProperties>> bindingServicePropertiesPerSbfb =
+							this.kafkaStreamsBindingInformationCatalogue.getConsumerPropertiesPerSbfb();
+					final List<ConsumerProperties> consumerProperties = bindingServicePropertiesPerSbfb.get(streamsBuilderFactoryBean);
+					final boolean autoStartupDisabledOnAtLeastOneConsumerBinding = consumerProperties.stream().anyMatch(consumerProperties1 -> !consumerProperties1.isAutoStartup());
+					if (!autoStartupDisabledOnAtLeastOneConsumerBinding) {
+						streamsBuilderFactoryBean.start();
+						this.kafkaStreamsRegistry.registerKafkaStreams(streamsBuilderFactoryBean);
```

Comment:
Don't we still need to register it, even if it is not started?

Reply:
We will get the undelrying `KafkaStreams` object only if we start the `StreamsBuilder`. Each time we stop/start, it is a new `KafkaStreams` object.

Reply:
Oh; I see that registration/unregistration is done in the binding start/stop too and it is idempotent since they are held in a set.

However, that set needs to be a `ConcurrentHashMap.newKeySet()` because the bindings can be started/stopped on different threads.